spock-core/src/main/java/org/spockframework/mock/EmptyOrDummyResponse.java

```diff
@@ -59,6 +74,10 @@ public Object respond(IMockInvocation invocation) {
       return ReflectionUtil.getDefaultValue(returnType);
     }
 
+    if (returnType.isAssignableFrom(invocation.getMockObject().getType())) {
+      return invocation.getMockObject().getInstance();
```

Comment:
Nice, one of the features available in Mockito 2. It could generate change a behavior of methods returning generic types (such as `Object`), however, I don't know if it is so often use case.